mailbox/backup/src/main/java/org/apache/james/mailbox/backup/DefaultMailboxBackup.java

```diff
@@ -81,8 +99,59 @@ public void backupAccount(User user, OutputStream destination) throws IOExceptio
         archive(mailboxes, messages, destination);
     }
 
-    private final MailboxManager mailboxManager;
-    private final ArchiveService archiveService;
+    @Override
+    public Publisher<Void> restore(User user, InputStream source) {
```

Comment:
IMO putting all the restore logic within that class makes it a bit too long. I'm suprised that such a high level class ends up doing some low-level mailbox manager calls...

Maybe we should extract this with the right granularity.

I think a MailArchiveRestorer (taking a MailArchiveEntry and then acting on the mailbox manager) might be a proposition.